src/Microsoft.Azure.ServiceBus/Core/MessageReceiver.cs

```diff
@@ -298,21 +300,35 @@ public Task<IList<Message>> ReceiveAsync(int maxMessageCount)
 
             MessagingEventSource.Log.MessageReceiveStart(this.ClientId, maxMessageCount);
 
+            bool isDiagnosticsEnabled = ServiceBusDiagnosticSource.IsEnabled();
```

Comment:
This is super nit-picky, I know: Consider either `areDiagnosticsEnabled` or `isDiagnosticsSourceEnabeld`. Why? Because "Diagnostics" are plural; whereas "DiagnosticsSource" is singular.

This might be a bit too contorted, but we might be able to avoid the problem altogether by eliminating the local `bool` and test for `activity != null` lower down.

Reply:
Thanks, I'm renaming to isDiagnosticSourceEnabled.

Regarding check for `actvity != null`,  while it works for Stop event, for exceptions, we assume some users may want to have Exception events only. With this level of verbosity there will be no activities created.